src/colvarproxy_io.cpp

```diff
@@ -0,0 +1,298 @@
+      std::string::npos) {
+    return rename_file(filename, (std::string(filename)+".old").c_str());
+  } else {
+    return rename_file(filename, (std::string(filename)+".BAK").c_str());
+  }
+}
+
+
+int colvarproxy_io::remove_file(char const *filename)
+{
+  int error_code = COLVARS_OK;
+#if defined(WIN32) && !defined(__CYGWIN__)
+  // Because the file may be open by other processes, rename it to filename.old
+  std::string const renamed_file(std::string(filename)+".old");
+  // It may still be there from an interrupted run, so remove it to be safe
+  std::remove(renamed_file.c_str());
+  int rename_exit_code = 0;
+  while ((rename_exit_code = std::rename(filename,
+                                         renamed_file.c_str())) != 0) {
+    if (errno == EINTR) continue;
+    error_code |= COLVARS_FILE_ERROR;
+    break;
+  }
+  // Ask to remove filename.old, but ignore any errors raised
+  std::remove(renamed_file.c_str());
+#else
+  if (std::remove(filename)) {
+    if (errno != ENOENT) {
+      error_code |= COLVARS_FILE_ERROR;
+    }
+  }
+#endif
+  if (error_code != COLVARS_OK) {
+    return cvm::error("Error: in removing file \""+std::string(filename)+
+                      "\".\n.",
+                      error_code);
+  }
+  return COLVARS_OK;
+}
+
+
+int colvarproxy_io::rename_file(char const *filename, char const *newfilename)
+{
+  int error_code = COLVARS_OK;
+#if defined(WIN32) && !defined(__CYGWIN__)
+  // On straight Windows, must remove the destination before renaming it
+  error_code |= remove_file(newfilename);
+#endif
+  int rename_exit_code = 0;
+  while ((rename_exit_code = std::rename(filename, newfilename)) != 0) {
+    if (errno == EINTR) continue;
+    // Call log() instead of error to allow the next try
+    cvm::log("Error: in renaming file \""+std::string(filename)+"\" to \""+
+             std::string(newfilename)+"\".\n.");
+    error_code |= COLVARS_FILE_ERROR;
+    if (errno == EXDEV) continue;
+    break;
+  }
+  return rename_exit_code ? error_code : COLVARS_OK;
+}
+
+
+std::istream *colvarproxy_io::input_stream(std::string const &input_name,
+                                           std::string const description,
+                                           bool error_on_fail)
+{
+  if (!io_available()) {
+    cvm::error("Error: trying to access an input file/channel "
+               "from the wrong thread.\n", COLVARS_BUG_ERROR);
+    return NULL;
+  }
+  std::istream *is = get_input_stream(input_name);
+  if (is != NULL) {
+    return is;
+  }
+
+  // Using binary to work around differences in line termination conventions
+  // See https://github.com/Colvars/colvars/commit/8236879f7de4
+  std::ifstream *isf = new std::ifstream(input_name.c_str(), std::ios::binary);
+
+  if (!isf->is_open()) {
+    if (error_on_fail) {
+      cvm::error("Error: cannot open "+description+" \""+input_name+"\".\n",
+                 COLVARS_FILE_ERROR);
+    }
+    return NULL;
+  }
+
+  input_streams_[input_name] = isf;
```

Comment:
The input stream is created here, but not deleted in the destructor. Is this a memory leak?